crates/provider/src/provider/multicall/mod.rs

```diff
@@ -107,6 +107,19 @@ where
             input_kind: TransactionInputKind::default(),
         }
     }
+
+    /// Converts an empty [`MulticallBuilder`] into a dynamic one
+    pub fn dynamic<D: SolCall + 'static>(self) -> MulticallBuilder<Dynamic<D>, P, N> {
+        MulticallBuilder {
+            calls: self.calls,
+            provider: self.provider,
+            block: self.block,
+            state_override: self.state_override,
+            address: MULTICALL3_ADDRESS,
+            input_kind: self.input_kind,
+            _pd: Default::default(),
+        }
+    }
 }
 
 impl<D: SolCall + 'static, P, N> MulticallBuilder<Dynamic<D>, P, N>
```

Comment:
can you also change `fn new_dynamic` to use this new method? `MulticallBuilder::new().dynamic()`